zeroconf/__init__.py

```diff
@@ -2930,23 +2963,24 @@ def send(self, out: DNSOutgoing, addr: Optional[str] = None, port: int = _MDNS_P
     def close(self) -> None:
         """Ends the background threads, and prevent this instance from
         servicing further queries."""
-        if not self._GLOBAL_DONE:
-            # remove service listeners
-            self.remove_all_service_listeners()
-            self.unregister_all_services()
-            self._GLOBAL_DONE = True
-
-            # shutdown recv socket and thread
-            if not self.unicast:
-                self.engine.del_reader(cast(socket.socket, self._listen_socket))
-                cast(socket.socket, self._listen_socket).close()
-            for s in self._respond_sockets:
-                self.engine.del_reader(s)
-            self.engine.join()
+        if self._GLOBAL_DONE:
```

Comment:
Let's restore the previous indentation here, it makes the patch larger than needed.

Reply:
Reverted